apprating/src/main/java/com/mindera/skeletoid/apprating/controller/AppRatingController.kt

```diff
@@ -0,0 +1,137 @@
+package com.mindera.skeletoid.apprating.controller
+
+import android.content.Context
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponse
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponseCallback
+import com.mindera.skeletoid.apprating.job.AppRatingJobInitializer
+import com.mindera.skeletoid.apprating.store.AppRatingStore
+import com.mindera.skeletoid.apprating.utils.DateUtils
+import java.util.Date
+
+/**
+ * Controller responsible for handling the stored data and the dialog itself
+ */
+class AppRatingController {
+
+    private var countsPerTimeInterval: Pair<Int, Long>? = null
+    private var promptTimeInterval: Long? = null
+
+    /**
+     * Sets up the conditions to prompt the dialog.
+     *
+     * @param countsPerTimeInterval Pair<Int, Long> values with the maximum number of times the dialog can be prompt per time range (in days)
+     * @param promptTimeInterval Time distance between prompts (in days)
+     */
+    fun setupConditions(countsPerTimeInterval: Pair<Int, Long>, promptTimeInterval: Long) {
+        this.countsPerTimeInterval = countsPerTimeInterval
+        this.promptTimeInterval = promptTimeInterval
+    }
+
+    /**
+     * Updates the values in the store.
+     */
+    fun updateStore(context: Context) {
+        val store = AppRatingStore(context)
+        val today = Date()
+        store.lastTimePrompted = DateUtils.formatDate(today)
+
+        countsPerTimeInterval?.let { (_, range) ->
+            if (store.initialPromptDate.isEmpty() || DateUtils.daysBetween(
+                    today,
+                    DateUtils.parseDate(store.initialPromptDate)
+                ) > range
+            ) {
+                store.initialPromptDate = DateUtils.formatDate(today)
+                store.promptedCount = 0
+            } else {
+                store.promptedCount++
+            }
+        }
+    }
+
+    /**
+     * Checks if the data in the store matches the conditions and updates the values in the store.
+     *
+     * @param context Context
+     * @param dialogResultCallback Callback to handle default dialog response
+     * @return true if the conditions were met, and the dialog can be prompted.
+     */
+    fun promptDialog(
+        context: Context,
+        dialogResultCallback: AppRatingDialogResponseCallback? = null
+    ): Boolean {
+        val store = AppRatingStore(context)
+
+        return shouldPromptDialog(store).also { shouldPromptDialog ->
+            if (shouldPromptDialog) {
+                dialogResultCallback?.let {
+                    TODO("IMPLEMENT DEFAULT DIALOG")
+                }
```

Comment:
I still don't like this because it's another collateral effect but I don't have any ideas on how to tackle it right now. Am I being too picky? What do you guys think?

Reply:
This will actually disappear, i'm already changing it here